ui/src/main/java/com/kotcrab/vis/ui/layout/FlowGroup.java

```diff
@@ -0,0 +1,402 @@
+			} else {
+				childWidth = child.getWidth();
+				childHeight = child.getHeight();
+			}
+
+			//See if it fits this column but place at least one child in the first column!
+			if (y - childHeight >= 0 || !wasChildProcessed) {
+				//Fits into this column.
+				currentColumnWidth = Math.max(childWidth, currentColumnWidth);
+			} else {
+				//Start new column.
+				totalWidth += currentColumnWidth + spacing;
+				y = targetHeight;
+				currentColumnWidth = childWidth;
+			}
+
+			float heightIncrement = childHeight + spacing;
+
+			y -= heightIncrement;
+			totalHeight += heightIncrement;
+
+			maxChildWidth = Math.max(maxChildWidth, childWidth);
+			maxChildHeight = Math.max(maxChildHeight, childHeight);
+
+			wasChildProcessed = true;
+		}
+
+		//Handle last row (if at least one row exists).
+		if (wasChildProcessed) {
+			//Remove the last spacing that was added excessively.
+			totalHeight -= spacing;
+		}
+
+		//Handle last column (no final spacing).
+		totalWidth += currentColumnWidth;
+
+		//Store results.
+		minWidth = maxChildWidth;
+		minHeight = maxChildHeight;
+
+		layoutedWidth = totalWidth;
+		layoutedHeight = targetHeight;
+
+		relaxedHeight = totalHeight;
+
+		sizeInvalid = false;
+	}
+
+	@Override
+	public void layout () {
+		if (horizontal) {
+			layoutHorizontal();
+		} else {
+			layoutVertical();
+		}
+	}
+
+	protected void layoutHorizontal () {
+		//NOTE: Should children invalidate/validate as per contract?
+
+		computeSizeIfNeeded();
+
+		final float targetWidth = getWidth();
+		final float targetHeight = getHeight();
+
+		float x = 0;
+		float y = targetHeight;
+		float rowHeight = 0;
+
+		SnapshotArray<Actor> children = getChildren();
+		boolean wasChildProcessed = false;
+
+		//Layout the child; first upside down as total height is, as of yet, unknown
+		//(due to the dynamic width used) and will be determined during this run.
+		for (Actor child : children) {
+			float childWidth;
+			float childHeight;
+
+			if (child instanceof Layout) {
+				Layout layout = (Layout) child;
+
+				childWidth = layout.getPrefWidth();
+				childHeight = layout.getPrefHeight();
+
+				//Need to update size.
+				child.setSize(childWidth, childHeight);
+			} else {
+				childWidth = child.getWidth();
+				childHeight = child.getHeight();
+
+				//Child already at size.
+			}
+
+			//See if it fits this row but place at least one child in the first row!
+			if (x + childWidth <= targetWidth || !wasChildProcessed) {
+				//Fits into this row.
+				rowHeight = Math.max(childHeight, rowHeight);
+			} else {
+				//Start new row.
+				x = 0;
+				y -= rowHeight + spacing;
+				rowHeight = childHeight;
+			}
+
+			child.setPosition(x, y - childHeight);
+
+			x += childWidth + spacing;
+
+			wasChildProcessed = true;
+		}
+
+		//Did a best effort to fit into the specified size but it still did not work.
+		//Let the ancestors know in hopes that they resize the group.
+		if (getHeight() != layoutedHeight) {
+			invalidateHierarchy();
+		}
```

Comment:
Will this really work in a case like this? Is it done somewhere in libGDX layouts?